new(big.Rat).SetString("") causes panic: runtime error: index out of range [2] with length 2

### Summary 

new(big.Rat).SetString("") causes panic: runtime error: index out of range [2] with length 2

### Steps to reproduce the behavior

Build gosec v2.12.0 from source using go1.18 darwin/amd64

```shell
$ git clone https://github.com/securego/gosec.git
...
$ cd gosec
$ git checkout v2.12.0
HEAD is now at a9b0ef0a1136 chore(deps): update all dependencies (#822)
$ go build ./cmd/gosec && echo OK
OK
```

Minimally reproducing example

```shell
$ ls
conversion.go  go.mod
$ cat go.mod
module conversion

go 1.19
$ cat conversion.go
package conversion

import "math/big"

func Conversion() {
	new(big.Rat).SetString("")
}
$ go build ./... && echo OK
OK
$ /tmp/gosec/gosec ./...
[gosec] 2022/08/05 13:48:59 Including rules: default
[gosec] 2022/08/05 13:48:59 Excluding rules: default
[gosec] 2022/08/05 13:48:59 Import directory: /tmp/conversion
[gosec] 2022/08/05 13:48:59 Checking package: conversion
[gosec] 2022/08/05 13:48:59 Checking file: /tmp/conversion/conversion.go
panic: runtime error: index out of range [2] with length 2

goroutine 1 [running]:
github.com/securego/gosec/v2.GoVersion()
	/tmp/gosec/helpers.go:458 +0xee
github.com/securego/gosec/v2/rules.(*usingOldMathBig).Match(0xc0001fea40, {0x145d478, 0xc000894240}, 0x4?)
	/tmp/gosec/rules/math_big_rat.go:24 +0x45
github.com/securego/gosec/v2.(*Analyzer).Visit(0xc0001361c0, {0x145d478, 0xc000894240})
	/tmp/gosec/analyzer.go:477 +0x843
go/ast.Walk({0x145be60?, 0xc0001361c0?}, {0x145d478?, 0xc000894240?})
	/usr/local/go/src/go/ast/walk.go:51 +0x62
go/ast.Walk({0x145be60?, 0xc0001361c0?}, {0x145d630?, 0xc000708b40?})
	/usr/local/go/src/go/ast/walk.go:206 +0xa10
go/ast.walkStmtList({0x145be60, 0xc0001361c0}, {0xc000708b50?, 0x1, 0x1383100?})
	/usr/local/go/src/go/ast/walk.go:32 +0x91
go/ast.Walk({0x145be60?, 0xc0001361c0?}, {0x145d428?, 0xc0007e29f0?})
	/usr/local/go/src/go/ast/walk.go:234 +0x606
go/ast.Walk({0x145be60?, 0xc0001361c0?}, {0x145d6f8?, 0xc0007e2a20?})
	/usr/local/go/src/go/ast/walk.go:357 +0x165c
go/ast.walkDeclList({0x145be60, 0xc0001361c0}, {0xc000840400?, 0x2, 0x110?})
	/usr/local/go/src/go/ast/walk.go:38 +0x91
go/ast.Walk({0x145be60?, 0xc0001361c0?}, {0x145d6a8?, 0xc000854300?})
	/usr/local/go/src/go/ast/walk.go:366 +0xdea
github.com/securego/gosec/v2.(*Analyzer).Check(0xc0001361c0, 0xc000424f00)
	/tmp/gosec/analyzer.go:301 +0x545
github.com/securego/gosec/v2.(*Analyzer).Process(0xc0001361c0, {0x0, 0x0, 0x0}, {0xc0000711e0, 0x1, 0x3d?})
	/tmp/gosec/analyzer.go:219 +0x450
main.main()
	/tmp/gosec/cmd/gosec/main.go:408 +0xc39
```

This issue also affects golangci-lint

Oddly, downloading the pre-build version of gosec 2.12 works just fine

```sh
$ ~/Downloads/gosec_2.12.0/gosec ./...
[gosec] 2022/08/05 13:49:44 Including rules: default
[gosec] 2022/08/05 13:49:44 Excluding rules: default
[gosec] 2022/08/05 13:49:44 Import directory: /tmp/conversion
[gosec] 2022/08/05 13:49:44 Checking package: conversion
[gosec] 2022/08/05 13:49:44 Checking file: /tmp/conversion/conversion.go
Results:


[/tmp/conversion/conversion.go:6] - G113 (CWE-190): Potential uncontrolled memory consumption in Rat.SetString (CVE-2022-23772) (Confidence: LOW, Severity: HIGH)
    5: func Conversion() {
  > 6: 	new(big.Rat).SetString("")
    7: }



Summary:
  Gosec  : 2.12.0
  Files  : 1
  Lines  : 7
  Nosec  : 0
  Issues : 1

$ 
```

### gosec version
v2.12.0 built from source

### Go version (output of 'go version')
go version go1.19 darwin/amd64

### Operating system / Environment
Darwin FAN-C02DV9LAMD6R 21.6.0 Darwin Kernel Version 21.6.0: Sat Jun 18 17:07:25 PDT 2022; root:xnu-8020.140.41~1/RELEASE_X86_64 x86_64 i386 Darwin

### Expected behavior
Not panicing

### Actual behavior
panicing

--- a/cmd/gosec/main.go
+++ b/cmd/gosec/main.go
@@ -71,7 +71,7 @@ func (a *arrayFlags) Set(value string) error {
 }
 
 var (
-	//#nosec flag
+	// #nosec flag
 	flagIgnoreNoSec = flag.Bool("nosec", false, "Ignores #nosec comments when set")
 
 	// show ignored
@@ -80,7 +80,7 @@ var (
 	// format output
 	flagFormat = flag.String("fmt", "text", "Set output format. Valid options are: json, yaml, csv, junit-xml, html, sonarqube, golint, sarif or text")
 
-	//#nosec alternative tag
+	// #nosec alternative tag
 	flagAlternativeNoSec = flag.String("nosec-tag", "", "Set an alternative string for #nosec. Some examples: #dontanalyze, #falsepositive")
 
 	// output file
@@ -148,7 +148,7 @@ var (
 	logger *log.Logger
 )
 
-//#nosec
+// #nosec
 func usage() {
 	usageText := fmt.Sprintf(usageText, Version, GitTag, BuildDate)
 	fmt.Fprintln(os.Stderr, usageText)
@@ -173,12 +173,12 @@ func usage() {
 func loadConfig(configFile string) (gosec.Config, error) {
 	config := gosec.NewConfig()
 	if configFile != "" {
-		//#nosec
+		// #nosec
 		file, err := os.Open(configFile)
 		if err != nil {
 			return nil, err
 		}
-		defer file.Close() //#nosec G307
+		defer file.Close() // #nosec G307
 		if _, err := config.ReadFrom(file); err != nil {
 			return nil, err
 		}
@@ -253,11 +253,11 @@ func printReport(format string, color bool, rootPaths []string, reportInfo *gose
 }
 
 func saveReport(filename, format string, rootPaths []string, reportInfo *gosec.ReportInfo) error {
-	outfile, err := os.Create(filename) //#nosec G304
+	outfile, err := os.Create(filename) // #nosec G304
 	if err != nil {
 		return err
 	}
-	defer outfile.Close() //#nosec G307
+	defer outfile.Close() // #nosec G307
 	err = report.CreateReport(outfile, format, false, rootPaths, reportInfo)
 	if err != nil {
 		return err
@@ -337,7 +337,7 @@ func main() {
 
 	// Ensure at least one file was specified or that the recursive -r flag was set.
 	if flag.NArg() == 0 && !*flagRecursive {
-		fmt.Fprintf(os.Stderr, "\nError: FILE [FILE...] or './...' or -r expected\n") //#nosec
+		fmt.Fprintf(os.Stderr, "\nError: FILE [FILE...] or './...' or -r expected\n") // #nosec
 		flag.Usage()
 		os.Exit(1)
 	}
@@ -460,7 +460,7 @@ func main() {
 	}
 
 	// Finalize logging
-	logWriter.Close() //#nosec
+	logWriter.Close() // #nosec
 
 	exit(issues, errors, *flagNoFail)
 }

--- a/helpers.go
+++ b/helpers.go
@@ -34,8 +34,8 @@ import (
 // initialization only imports.
 //
 // Usage:
-// 	node, matched := MatchCallByPackage(n, ctx, "math/rand", "Read")
 //
+//	node, matched := MatchCallByPackage(n, ctx, "math/rand", "Read")
 func MatchCallByPackage(n ast.Node, c *Context, pkg string, names ...string) (*ast.CallExpr, bool) {
 	importedName, found := GetImportedName(pkg, c)
 	if !found {
@@ -474,9 +474,25 @@ func RootPath(root string) (string, error) {
 
 // GoVersion returns parsed version of Go from runtime
 func GoVersion() (int, int, int) {
-	versionParts := strings.Split(runtime.Version(), ".")
-	major, _ := strconv.Atoi(versionParts[0][2:])
-	minor, _ := strconv.Atoi(versionParts[1])
-	build, _ := strconv.Atoi(versionParts[2])
+	return parseGoVersion(runtime.Version())
+}
+
+// parseGoVersion parses Go version.
+// example:
+// - go1.19rc2
+// - go1.19beta2
+// - go1.19.4
+// - go1.19
+func parseGoVersion(version string) (int, int, int) {
+	exp := regexp.MustCompile(`go(\d+).(\d+)(?:.(\d+))?.*`)
+	parts := exp.FindStringSubmatch(version)
+	if len(parts) <= 1 {
+		return 0, 0, 0
+	}
+
+	major, _ := strconv.Atoi(parts[1])
+	minor, _ := strconv.Atoi(parts[2])
+	build, _ := strconv.Atoi(parts[3])
+
 	return major, minor, build
 }

